core/sqf/src/seatrans/tm/hbasetmlib2/src/main/java/org/trafodion/dtm/HBaseTxClient.java

```diff
@@ -1062,6 +1062,14 @@ public void run() {
                                               tmID + " regionBytes: [" + regionBytes + "].  Deleting zookeeper region entry. \n exception: " + de);
                                       zookeeper.deleteRegionEntry(regionEntry);
                                 }// DeserializationException
+                                catch (IOException ioe) {
+                                    // It's possible we received a spurious entry from a stale region.
+                                       // It's safe to delete the zookeeper entry and wait for any new posts
+                                       LOG.warn("TRAF RCOV THREAD:IOException calling txnManager.recoveryRequest. " + "TM: " +
+                                               tmID + " regionBytes: [" + regionBytes
+                                               + "].  Deleting zookeeper region entry. exception: " + ioe);
+                                       zookeeper.deleteRegionEntry(regionEntry);
+                                 }// IOException
```

Comment:
As per of cleaning up the exception handling on the client , I attempted to do the following for the coprocessorService request.

 CoprocessorService request throws either ServiceException or Throwable.  ServiceException  are flagged as RetryTransactionException and Throwable are flagged as DoNotRetryException.  

When there is a response, the exceptionString will be looked at and converted into an exception derived from IOException.  If some of these exceptions can be retried, it would be flagged as RetryTransactionException. The RetryTransactionException is retried 'n' numbers of time, before it is thrown as an DoNotRetryException.   

This concept is missed out for recoveryRequest.  It would be good if we can add this concept for recoveryRequest. 

By ignoring the IOException here, I am wondering if we ignoring some exceptions which should not be ignored.  The other exceptions like TableNotFoundException are getting masked here as IOException because recoveryRequest throws only IOException. The tableNotFoundException could be the cause of the IOException.  In that case, we should catch IOException and get the cause of the exception and handle the causes here.